From e1af176af2bb274745e9b0d43456763f8946f107 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@linaro.org>
Date: Mon, 25 May 2020 13:09:59 +0200
Subject: [PATCH] xtest: fix generic argument handling

Fix position of last generic argument badly handled that makes -d, -l or
-t argument being treated also as if it was the first test ID to consider
in the filtered test IDs list.

Fixes: 66542a1bb4e0 ("xtest: exclude tests with -x")
Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Acked-by: Jerome Forissier <jerome@forissier.org>
---
 host/xtest/xtest_main.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/host/xtest/xtest_main.c b/host/xtest/xtest_main.c
index 3dff2fe..1e566fa 100644
--- a/host/xtest/xtest_main.c
+++ b/host/xtest/xtest_main.c
@@ -188,8 +188,6 @@ int main(int argc, char *argv[])
 		}
 	}
 next:
-	if (last_gen_option > 1)
-		last_gen_option--;
 
 	for (index = last_gen_option; index < argc; index++) {
 		if (!strcmp(argv[index], "-x")) {
-- 
2.17.1

